### DISCUSSION TITLE
trading with stop order within a candle

### DISCUSSION THREAD

**Discussion by shovalyechiel** [kernc/backtesting.py#785] [Q&A]
Hey everyone! 
Been playing around with this great engine and have some very nice results and insights so far!
I have an idea for some ( very ) short term plays following 1 min candles and have some very small sl/tp (about .5% size). i wanted to run my strategy with stop-orders but i have some problems.
For this example i set the commision to be 0.
1) Lets say i open on candle 1 a long-stoporder at the price of 100$ and candle 1 closes at 99$

2) Candles 2 assuming opens on 99 as well and closes on 101$. at that point my long stoporder will be triggered right? but from what i see in my trades (and the code ) the entry price would be 101$ (candle Close ). so in this specific case this is a 1% off.

i also tried to do stop + limit (but as i suspected from viewing the code it still doesn't give me the wanted results.

So as far as i can tell the way im testing currently can give me some very misleading results from the deviation. 
and as what i think when a stop orders gets hit it feels like it should be in the stoporder price itself but maybe im missing something here ?
any idea how i can run some more accurate test with in-candle trading ?

Thanks!

**Comment by MarcosScherer** [comment]
Hi everyone!
Problem Ex: Buy index in 12.000 points, stop in 11.980 points.
Next candle close: 11.970.
The algo stop in 11.970 but passed my stop.
I want the algo to stop in 11.980.

If you can help...
Thanks a lot

**Reply by schoeth** [reply]
That works, only the 2nd two trades with the gaps are a bug, which is already reported:

```
import pandas as pd
from backtesting import Strategy, Backtest

data = {
    'Open' : [12000, 12000, 12000, 11972, 11970, 12000, 11985],
    'High' : [12000, 12000, 12000, 12000, 11970, 12000, 11990],
    'Low'  : [12000, 12000, 11970, 11990, 11970, 12000, 11980],
    'Close': [12000, 12000, 11970, 12000, 11970, 12000, 11985],
}
dataframe = pd.DataFrame(data=data)

class StoplossTest(Strategy):
    def init(self):
        pass
    def next(self):
        if self.data.Close[-1] == 12000:
            self.buy(size = 1, sl = 11980)
        elif self.data.Close[-1] == 11970:
            self.sell(size = 1, sl = 11990)
    
bt  = Backtest(dataframe, StoplossTest, cash = 62000, trade_on_close = True)
opt = bt.run()
bt.plot()
```
